checker/src/abstract_value.rs

```diff
@@ -2565,8 +2565,14 @@ impl AbstractValueTrait for Rc<AbstractValue> {
             | Expression::HeapBlockLayout { .. }
             | Expression::Reference { .. }
             | Expression::UnknownModelField { .. }
-            | Expression::Variable { .. } => {
-                return TagDomain::for_empty_set();
+            | Expression::UnknownTagCheck { .. } => return TagDomain::empty_set(),
```

Comment:
The underlying value may actually have tags, just not the one this check is looking for. Better return those.

Reply:
This depends on "whether a tag check can propagate other tags or not". What's your opinion here?  Should we make this controllable via tag propagation set?

Reply:
Thinking about this some more, I can't convince myself that a compile time only operation need to be taint checked. So, I think I now agree with the code as you wrote it.